ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStreamBase.java

```diff
@@ -205,14 +228,19 @@ private void runTestDataStream(int bufferSize, int bufferNum) {
       Assert.assertEquals(reply.getType(), Type.STREAM_DATA);
     }
 
-    for (SingleDataStreamStateMachine s : singleDataStreamStateMachines) {
-      RaftClientRequest writeRequest = s.getWriteRequest();
-      if (writeRequest.getClientId().equals(impl.getHeader().getClientId())) {
-        Assert.assertEquals(writeRequest.getCallId(), impl.getHeader().getCallId());
-        Assert.assertEquals(writeRequest.getRaftGroupId(), impl.getHeader().getRaftGroupId());
-        Assert.assertEquals(writeRequest.getServerId(), impl.getHeader().getServerId());
+    final RaftClientRequest header = out.getHeader();
+    for (MultiDataStreamStateMachine s : stateMachines) {
+      final SingleDataStream stream = s.getSingleDataStream(header.getCallId());
+      if (stream == null) {
+        continue;
+      }
```

Comment:
Maybe we can `Assert.assertNotNull(stream)` instead of `continue `?

Reply:
Assert.assertNotNull(stream) will fail the test when stream == null. It is a possible case when there are multiple servers.